project-koku/koku
# [COST-311] Cost Model costs remain after deleting cost model

When a cost_model was deleted from a source, it would skip updating the cost models because it was not associated with a source. Now when a model is deleted it will update the cost models and refresh the materialized views.

Some Notes:
- Changed update_cost_model_costs task to no longer check if a cost model exists and just run the update, moved the check to update_summary_tables. Moved because when a cost_model was deleted, it would not update cost_model_costs since there was no cost_model.
- Refactored update_summary_tables for more logical flow

Testing:
1. Add an azure source
2. Add a cost model to that source:
```
{
    "name": "Test Cost Model",
    "description": "Test",
    "source_type": "Azure",
    "source_uuids": ["{uuid}"],
    "markup": {"value": 10, "unit": "percent"}
}
```
3. Check the costs endpoint: [http://127.0.0.1:8000/api/cost-management/v1/reports/azure/costs/](http://127.0.0.1:8000/api/cost-management/v1/reports/azure/costs/)
4. Delete the cost model
5. Check the costs endpoint again and see that the markup is 0.
